server/speech.js

```diff
@@ -0,0 +1,119 @@
+    SELECT transcript
+    FROM evidence
+    WHERE 1=1
+      AND json->>'downloadUrl' = $1
+    ORDER BY json->>'sessionId', id ASC;`;
+
+  pool.query(sql,values)
+    .then(results => {
+      if ((results.rowCount >= 1) && (results.rows[0].transcript !== "")){
+        console.log('found transcript: ', results.rows[0].transcript);
+        response.set('Content-Type', 'application/json');
+        response.json({
+          transcript: results.rows[0].transcript
+        });
+        return response.status(200).end();
+      }
+      else{
+        console.log('need to transcribe');
+        speechToText(pool,audioFile)
+          .then(results => {
+            console.log('final transcript: ',results);
+            response.set('Content-Type', 'application/json');
+            response.json({
+              transcript: results 
+            });
+            return response.status(500).end();
+          })
+          .catch(err => {
+            console.log('Error with transcription:', err);
+            return response.status(500).end();
+          });
+      }
+    })
+    .catch(err => {
+      console.log('Error with database query for transcript:', err);
+      return response.status(500).end();
+    });
+}
+
+// Sets up connection to IBM Watson api and transcribes audio to text.
+// returns a promise resolved with the transcript
+function speechToText(pool, audioFile) {
+  //changing audioFile here to a wav file on my local disk to transcribe
+  //Should not need this line when hooked up to s3
+  audioFile = '/Users/keving17/Documents/Github/TSL/teacher-moments/server/20180213-185953-485_8d44f67e-d783-4597-aec2-d417325cce06.wav';
+  
+  var SpeechToTextV1 = require('watson-developer-cloud/speech-to-text/v1');
+  var fs = require('fs');
+
+  var watson = new SpeechToTextV1 ({
+    username: '54aa1588-bbee-4e5e-8cb9-0533e4494d91',
+    password: 'eR8ayEdlzCL4'
+  });
+  
+  var params = {
+    audio: fs.createReadStream(audioFile),
+    content_type: 'audio/wav',                          // eslint-disable-line camelcase
+    timestamps: true,
+    word_alternatives_threshold: 0.9,                   // eslint-disable-line camelcase
+    keywords: ['colorado', 'tornado', 'tornadoes'],
+    keywords_threshold: 0.5                             // eslint-disable-line camelcase
+  };
+  
+  return new Promise((resolve,reject) => {
+    watson.recognize(params, function(error, transcript) {
+      if (error) {
+        console.log('Could not transcribe with IBM Watson:', error);
+        reject('Could not transcribe with IBM Watson:', error);
+      }
+      else{
+        const text = transcript.results[0].alternatives[0].transcript;
+        updateTranscript(pool, text, audioFile)
+          .then(results => {
+            resolve(results);
+          });
+      }
+    });
+  });
+}
+
+// Updates database with transcript so we dont need to transcribe 
+// the same wav files multiple times
+function updateTranscript(pool, text, audioFile) {
```

Comment:
Yeah this is awesome!  So in regards to my comment #3 above around how the engineering design works, the way you wrote this code in different functions is great.  You could shuffle it around and re-compose it into different designs, depending on the tradeoffs you want in terms of whether all past data is transcribed, how different engineering designs impact the researcher user experience, etc.

Right now, the work to do the transcription is on the "read path" - when the user asks for a response, the server goes and computes the transcription then, on-demand, if it's needed.  Another design could move this to the "write path," where whenever the server receives new audio evidence, it writes it to the database, and runs the transcription then.  Or another design is that could move this to an "offline job" that does the work in batches periodically for all the recent audio evidence.

Anyway, I think focusing on #1 transcription quality and #2 data sharing are best to start with, but just calling out that the way you wrote this proof-of-concept is great for then iterating on the final design that gets the tradeoffs you want for cost / user experience / etc.